event_contact/tests/test_event.py

```diff
@@ -1,35 +1,55 @@
-# -*- coding: utf-8 -*-
 # Copyright 2016-2017 Jairo Llopis <jairo.llopis@tecnativa.com>
+# Copyright 2019 Alexandre Díaz <alexandre.diaz@tecnativa.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
+from datetime import datetime,  timedelta
 from openerp.tests.common import TransactionCase
 
 
 class EventCase(TransactionCase):
     def setUp(self, *args, **kwargs):
         super(EventCase, self).setUp(*args, **kwargs)
-        self.type1 = self.env.ref("event.event_type_1")
-        self.type2 = self.env.ref("event.event_type_2")
-        self.type1.contact_ids = (
-            self.env.ref("base.res_partner_1") |
-            self.env.ref("base.res_partner_2"))
-        self.type2.contact_ids = (
-            self.env.ref("base.res_partner_3") |
-            self.env.ref("base.res_partner_4"))
 
-        self.event1 = self.env.ref("event.event_1")
+        # Partners
+        self.partner1 = self.env['res.partner'].create({
+            'name': 'Test Partner 1',
+        })
+        self.partner2 = self.env['res.partner'].create({
+            'name': 'Test Partner 2',
+        })
+        self.partner3 = self.env['res.partner'].create({
+            'name': 'Test Partner 3',
+        })
+        self.partner4 = self.env['res.partner'].create({
+            'name': 'Test Partner 4',
+        })
+
+        self.type1 = self.env['event.type'].create({
+            'name': 'Event Type Test 1',
+            'contact_ids': [(6, False, [self.partner1.id, self.partner2.id])],
+        })
+        self.type2 = self.env['event.type'].create({
+            'name': 'Event Type Test 2',
+            'contact_ids': [(6, False, [self.partner3.id, self.partner4.id])],
+        })
+
+        self.event1 = self.env['event.event'].create({
+            'name': 'Event Test 1',
+            'date_begin': datetime.now(),
+            'date_end': datetime.now() + timedelta(days=3),
+        })
 
     def test_event_onchange_type_contacts_empty(self):
         """You get default contacts from type."""
-        self.event1.event_type_id = self.type2
+        self.event1.event_type_id = self.type2.id
```

Comment:
Why do need to assign like this?

Reply:
In Odoo the two forms are used interchangeably... i like use 'id' because internally uses this value to set the field.